### PR TITLE
Use etcdv3 by default

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description
A few sentences describing the overall goals of the pull request's commits.
Please include
- the type of fix - (e.g. bug fix, new feature, documentation)
- some details on _why_ this PR should be merged
- the details of the testing you've done on it (both manual and automated)
- which components are affected by this PR

## Todos
- [ ] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: config/config_params.go [hunk 1]
```diff
@@ -357,9 +357,9 @@ func (config *Config) resolve() (changed bool, err error) {
 }
 
 func (config *Config) DatastoreConfig() apiconfig.CalicoAPIConfig {
-	// Special case for etcdv2 datastore, where we want to honour established Felix-specific
+	// Special case for etcd3 datastore, where we want to honour established Felix-specific
```
Comment (bcreane) [inline config/config_params.go]: etcd3 -> etcdv3

**Comment by bcreane** [comment]
ugh, looks like many of the tests that I ran last night successfully against the incorrect `etcdv2` are now failing

**Comment by neiljerram** [comment]
@bcreane Which tests were those?  I think that libcalico-go doesn't support etcdv2 anymore, so not sure how it could be possible for you to run with etcdv2 by accident.  (Also note that the k8sfv test has
```
       -e CALICO_DATASTORE_TYPE=kubernetes \
       -e FELIX_DATASTORETYPE=kubernetes \
```
as it always has had, and that the Felix FV tests have (since the v3 merge work last week)
```
		"-e", "CALICO_DATASTORE_TYPE=etcdv3",
		"-e", "FELIX_DATASTORETYPE=etcdv3",
```
so - at least in these tests - the default doesn't actually come into play.)

**Comment by bcreane** [comment]
@neiljerram - looked like half a dozen k8sfv's were failing on semaphore, but I haven't checked recently.

**Comment by neiljerram** [comment]
@bcreane Half a dozen is worse than I would expect.  I believe there's a flake with the "pod whose IP has been cleared" test, depending on which tests run before it - but that is just one test.

(To be sure though, did you really mean 'k8sfv' tests here, as opposed to tests in the Felix 'fv' suite?)

**Comment by caseydavenport** [comment]
nit has been fixed, hopefully the tests pass!

**Comment by neiljerram** [comment]
Unfortunately tests still failing here. I'm going to investigate why.

**Comment by neiljerram** [comment]
Got it.  With this change, Felix container needs to be run with
```
		"-e", "FELIX_ETCDENDPOINTS=http://"+etcdIP+":2379",
```
instead of (or as well as)
```
		"-e", "CALICO_ETCD_ENDPOINTS=http://"+etcdIP+":2379",
```

**Comment by neiljerram** [comment]
Continued in #1582

**Pull request closed without merging**